pkg/factories/rune_factory.go

```diff
@@ -220,6 +223,58 @@ func (f *kafkaSourceRunEFactory) DescribeRunE() sourcetypes.RunE {
 	}
 }
 
+func (f *kafkaSourceRunEFactory) ListRunE() sourcetypes.RunE {
+	return func(cmd *cobra.Command, args []string) error {
+		var err error
+		namespace, err := f.KnSourceParams().GetNamespace(cmd)
+		if err != nil {
+			return err
+		}
+		restConfig, err := f.KnSourceParams().KnParams.RestConfig()
+		if err != nil {
+			return err
+		}
+
+		f.kafkaSourceClient, err = f.KafkaSourceClient(restConfig, namespace)
+		if err != nil {
+			return err
+		}
+
+		kafkaSourceList, err := f.kafkaSourceClient.GetKafkaSources()
+
+		if err != nil {
+			return fmt.Errorf(
+				"cannot list KafkaSources in namespace '%s' "+
+					"because: %s", f.kafkaSourceClient.Namespace(), err)
+		}
+		if len(kafkaSourceList.Items) == 0 {
+			return fmt.Errorf("no kafka source found in namespace '%s'", f.kafkaSourceClient.Namespace())
+		}
+
+		printer := printers.NewTableGenerator()
+		kafkaSourceListColumnDefinitions := []metav1.TableColumnDefinition{
+			{Name: "Name", Type: "string", Description: "Name of the created kafka source", Priority: 1},
+			{Name: "Age", Type: "string", Description: "Age of the kafka source", Priority: 1},
+			{Name: "BootstrapServers", Type: "string", Description: "Kafka bootstrap server", Priority: 1},
```

Comment:
Not sure if this is possbile, but I can think that the list of bootstrap servers can be quite large as a column. Is there a way to truncate this to a max. nr of chars ?

Same for topics. Also I wouldn't add _every_ field of a source as a user can always use `describe` for more details. While I think that the bootstrap servers might be important to distinguish different servers, I don't think we should include the consumergroup by default, and maybe even not the topic. 

Not sure about the Age, but this might be helpful, too (i would check whether other list commands include the age, too, as the Age is probably available everywhere)

I might even prefer to list the _sink_ instead (in a short form), as this is an important information whether a source is actively sending events, or not.

So in summary:

* Remove Topics & ConsumerGroup, shorten bootstrap serves to e.g. 30 chars. Add Age only if other sources add them, too.
* Add sink in a short form, if this possible. Again this should be kind of aligned with other sources (don't know it by heart how it looks like for the other sources though